### ISSUE TITLE
Vector layer mouseleave not triggered when leaving quickly

### ISSUE THREAD

**Issue by program-sam** [openglobus/openglobus#763]
I noticed some weird behavior on the mouseleave event of a vector layer. The event seem to trigger correctly, unless you leave the vector more quickly, after which the event never triggers.

![ezgif com-optimize](https://github.com/openglobus/openglobus/assets/75586344/d17c7a1f-114b-49a7-8f4c-e6122acb808a)

**Comment by Zemledelec** [comment]
I'll check...

**Comment by Zemledelec** [comment]
...can't reproduce. Could you show the code, please...or, can I take a look somewhere? Looks like fps effects, but I cant say for sure by the video.

**Comment by Zemledelec** [comment]
...also, can you check also, but first question do you use openglobus as a submodule or npm?

**Comment by program-sam** [comment]
I took code inspired from the Vector Layer Example:

```
  boundsLayer.events.on("mouseleave", function (e) {
    e.pickingObject.geometry.setFillColor(0, 0, 0, 0.1);
    e.renderer.handler.canvas.style.cursor = "default";
  });

  boundsLayer.events.on("mouseenter", function (e) {
    e.pickingObject.geometry.bringToFront();
    e.pickingObject.geometry.setFillColor(0.2, 0.2, 1, 0.4);
    e.renderer.handler.canvas.style.cursor = "pointer";
  });
```

Disabling the line that changes the cursor fixes the issue. Very weird but It would still be good to figure out why changing the cursor has this effect on the mouseleave event.

I am using openglobus as npm package. I can't reproduce it in the vectorLayer example by adding changing the cursor. so that is very weird. I'll let you know when I manage to make something reproduceable.

edit: cursor does not seem to be the full story, It seemed to fix it at first, but I still encounter the issue regardless of the cursor. There also don't seem to be dropped frames which could miss the event

**Comment by Zemledelec** [comment]
@program-sam could you use a submodule? In this case, I could ask you to make some things inside the engine. There is a case when the renderer stops dispatching any events (for optimization purposes), and we could disable it.

**Comment by Zemledelec** [comment]
...also, try to remove bringToFront(), but nevertheless, I cant reproduce the bug. 

What about Chrome?

P.S.
Also, I noticed that `mouseleave` doesn't happen when you move the mouse to HTML elements, like buttons, even out of the screen. Could be that something browser rendering dom elements affects this issue...I'll fix this one after the new release with typescript, but It's just a guess, about browser dom rendering.

**Comment by program-sam** [comment]
Oh cool, I hadn't heard of submodules, but seems to be interesting! I'll use that for openglobus from now on. will make testing easier. It seems to be across browsers. One difference is that I have a mini-map in the corner. A second globus inside another canvas. disabling this seems to solve it! I have no clue why that would cause the mouseleave to _sometimes_ not trigger? Does this give you any idea?

**Comment by Zemledelec** [comment]
I suppose, some browser issue, not actually  a bug, but a conflict, I cant say why. I ll try to make a mouseleave triggered when lost positioning target, it could solve.


Btw, I strongly recommend using a submodule.

**Comment by Zemledelec** [comment]
Second globus you mean the globus instance?

**Comment by Zemledelec** [comment]
I think for such mini map it's better to not to use another globus, it's too heavy. it's better to write a plugin in the one globus context. I'll think about it. Nevertheless, I'll check your case.

**Comment by program-sam** [comment]
> I think for such mini map it's better to not to use another globus, it's too heavy. it's better to write a plugin in the one globus context. I'll think about it. Nevertheless, I'll check your case.

ok! Minimap is definitely much lighter than the main planet (only 1 tile layer), but perhaps a secondary camera at a certain z-offset could be better for performance. I'll make a discussion post about it when I get to it

**Comment by Zemledelec** [comment]
Hi @program-sam, could you check the last master?

Beware it's typescript now.

**Comment by Zemledelec** [comment]
@program-sam need help?

**Comment by program-sam** [comment]
Thanks for the check-in! I did indeed had some issues with using openglobus as a sub-module, especially with the typescript implementation. But I did get to test it with the latest release and the changes you made seem to have solved the issue perfectly!

![ezgif com-video-to-gif (1)](https://github.com/openglobus/openglobus/assets/75586344/4de7e2bf-b9ee-473c-bd2e-c22a0faf5cd4)

Even though I did end up using another implementation to fix the events not passing through layers (#762)
